core/src/test/java/org/apache/iceberg/view/TestViewRepresentationParser.java

```diff
@@ -0,0 +1,177 @@
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot deserialize unknown view representation type non-sql");
+  }
+
+  @Test
+  public void testParseSqlViewRepresentation() {
+    String requiredFields =
+        "{\"type\":\"sql\", \"sql\": \"select * from foo\", \"schema-id\": 1, \"dialect\": \"spark-sql\"}";
+    List<Schema> schemas = ImmutableList.of(new Schema(1));
+    SQLViewRepresentation viewRepresentation =
+        ImmutableSQLViewRepresentation.builder()
+            .query("select * from foo")
+            .schema(new Schema(1))
+            .dialect("spark-sql")
+            .build();
+    validateSqlRepresentation(
+        viewRepresentation, ViewRepresentationParser.fromJson(requiredFields, schemas));
+
+    String requiredAndOptionalFields =
+        "{\"type\":\"sql\", \"sql\": \"select * from foo\", \"schema-id\": 1, \"dialect\": \"spark-sql\", "
+            + "\"default-catalog\":\"cat\", "
+            + "\"default-namespace\":[\"part1\",\"part2\"], "
+            + "\"field-aliases\":[\"col1\", \"col2\"], "
+            + "\"field-comments\":[\"Comment col1\", \"Comment col2\"]}";
+
+    SQLViewRepresentation viewWithOptionalFields =
+        ImmutableSQLViewRepresentation.builder()
+            .query("select * from foo")
+            .schema(new Schema(1))
+            .dialect("spark-sql")
+            .defaultCatalog("cat")
+            .fieldAliases(ImmutableList.of("col1", "col2"))
+            .fieldComments(ImmutableList.of("Comment col1", "Comment col2"))
+            .defaultNamespace(Namespace.of("part1", "part2"))
+            .build();
+
+    validateSqlRepresentation(
+        viewWithOptionalFields,
+        ViewRepresentationParser.fromJson(requiredAndOptionalFields, schemas));
+  }
+
+  private void validateSqlRepresentation(
+      SQLViewRepresentation expected, ViewRepresentation actual) {
+    Assert.assertEquals("Expected type sql", expected.type(), actual.type());
+
+    SQLViewRepresentation sqlViewRepresentation = (SQLViewRepresentation) actual;
+    Assert.assertEquals(
+        "Expected same query string", expected.query(), sqlViewRepresentation.query());
+    Assert.assertEquals(
+        "Expected same dialect", expected.dialect(), sqlViewRepresentation.dialect());
+    Assert.assertTrue(expected.schema().sameSchema(sqlViewRepresentation.schema()));
+    Assert.assertEquals(
+        "Expected same default namespace",
+        expected.defaultNamespace(),
+        sqlViewRepresentation.defaultNamespace());
+    Assert.assertEquals(
+        "Expected same default catalog",
+        expected.defaultCatalog(),
+        sqlViewRepresentation.defaultCatalog());
+    Assert.assertEquals(
+        "Expected same field aliases",
+        expected.fieldAliases(),
+        sqlViewRepresentation.fieldAliases());
+    Assert.assertEquals(
+        "Expected same field comments",
```

Comment:
nit: I think mentioning that we're expecting the same XYZ is probably not needed in all of these `assertEquals()` calls